getpfp.py

```diff
@@ -0,0 +1,75 @@
+# -*- coding: future_fstrings -*-
+
+#    Friendly Telegram (telegram userbot)
+#    Copyright (C) 2018-2019 The Authors
+
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as published by
+#    the Free Software Foundation, either version 3 of the License, or
+#    (at your option) any later version.
+
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import logging
+
+from .. import loader, utils
+
+logger = logging.getLogger(__name__)
+
+
+def register(cb):
+    cb(GetPPMod())
+
+
+class GetPPMod(loader.Module):
+    """Gets the profile photos of the target"""
+    def __init__(self):
+        self.name = _("Profile Photo Fetcher")
+
+    async def client_ready(self, client, db):
+        self.client = client
+        self.db = db
+
+    async def getpfpcmd(self, message):
+        """Fetches the profile photos of replied users, channels or chats"""
+        id = utils.get_args_raw(message)
+        user = await message.get_reply_message()
+        chat = message.input_chat
+        if user:  # If replied to a message, photos should be extracted from the user
+            photos = await self.client.get_profile_photos(user.sender)
+        else:
+            photos = await self.client.get_profile_photos(chat)
+        if not id:  # if photo id isn't stated, it will fetch all the photos of target
+            if len(photos) > 0:
+                await self.client.send_file(message.chat_id, photos)
+            else:
+                try:  # I had to do this bit because get_profile_photos function wasnt working for one photo
+                    if user:
+                        photo = await self.client.download_profile_photo(user.sender)
+                    else:
+                        photo = await self.client.download_profile_photo(message.input_chat)
+                    await self.client.send_file(message.chat_id, photo)
+                except Exception:
+                    await message.edit("<b>This user has no photos.</b>")
+                    return
+        else:  # If photo id is stated
```

Comment:
If you only want one, why does the above code download all of them? Take, for example, t.me/spechide. There's almost 200000 profile photos. Imagine downloading all of those, just to access one of them.